enterprise/reporting/reporting.go

```diff
@@ -394,7 +397,13 @@ func (r *DefaultReporter) mainLoop(ctx context.Context, c types.SyncerConfig) {
 				r.log.Infof("stopping mainLoop for syncer %s : %s", c.Label, ctx.Err())
 				return ctx.Err()
 			}
-			requestChan := make(chan struct{}, r.maxConcurrentRequests.Load())
+
+			concurrentRequests := r.maxConcurrentRequests.Load()
+			if time.Since(lastReportedAtTime.Load()).Minutes() > 10 {
+				concurrentRequests = r.maxConcurrentRequestsWithDelay.Load()
+			}
+
+			requestChan := make(chan struct{}, concurrentRequests)
```

Comment:
It would be useful, to introduce a metric here. I would suggest a gauge for `concurrentRequests`.

Is there a metric for `time.Since(lastReportedAtTime.Load())` ?

Reply:
Yes we have a metrics for `time.Since(lastReportedAtTime.Load())`